object/check.go

```diff
@@ -293,29 +345,81 @@ func CheckUserPassword(organization string, username string, password string, la
 		panic(err)
 	}
 
-	if user == nil || user.IsDeleted {
-		return nil, fmt.Sprintf(i18n.Translate(lang, "general:Invalid username or password/code"))
+	if user == nil {
+		return nil, NewCheckUserPasswordError(ErrorUserNotFound)
+	}
+
+	if user.IsDeleted {
+		return nil, NewCheckUserPasswordError(ErrorUserDeleted)
 	}
 
 	if user.Ldap != "" {
 		// ONLY for ldap users
 		if msg := checkLdapUserPassword(user, password, lang); msg != "" {
 			if msg == "user not exist" {
-				return nil, fmt.Sprintf(i18n.Translate(lang, "general:Invalid username or password/code"))
+				return nil, NewCheckUserPasswordError(ErrorLDAPUserNotFound)
 			}
-			return nil, msg
+			return nil, NewCheckUserPasswordError(ErrorLDAPError).WithMessage(msg)
 		}
 	} else {
 		if msg := CheckPassword(user, password, lang, enableCaptcha); msg != "" {
-			return nil, msg
+			return nil, NewCheckUserPasswordError(ErrorWrongPassword).WithMessage(msg)
 		}
 	}
 
 	if user.IsForbidden {
-		return nil, i18n.Translate(lang, "check:The user is forbidden to sign in, please contact the administrator")
+		return nil, NewCheckUserPasswordError(ErrorUserBlocked)
 	}
 
-	return user, ""
+	return user, nil
+}
+
+func LogRecordCheckPassError(ctx *context.Context, err error) {
+	if extendedErr, ok := err.(*CheckUserPasswordError); ok {
+		switch extendedErr.Err() {
+		case ErrorUserNotFound:
+			record := NewRecordBuilder(ctx).WithDetail("User not found").Build()
+			SaveOnSuccess(ctx, record)
+		case ErrorUserDeleted:
+			record := NewRecordBuilder(ctx).WithDetail("User deleted").Build()
+			SaveOnSuccess(ctx, record)
+		case ErrorUserBlocked:
+			record := NewRecordBuilder(ctx).WithDetail("User blocked").Build()
+			SaveOnSuccess(ctx, record)
+		case ErrorWrongPassword:
+			record := NewRecordBuilder(ctx).WithDetail("Wrong password").Build()
+			SaveOnSuccess(ctx, record)
+		case ErrorLDAPUserNotFound:
+			record := NewRecordBuilder(ctx).WithDetail("LDAP user not found").Build()
+			SaveOnSuccess(ctx, record)
+		case ErrorLDAPError:
+			record := NewRecordBuilder(ctx).WithDetail("LDAP error").Build()
+			SaveOnSuccess(ctx, record)
+		}
```

Comment:
NewRecordBuilder(ctx).WithDetail(err.Error()).Build() isn't enough here instead switch statement?

Reply:
it's just a mapping errors to only needed events and it description

Reply:
but all detail I see here exactly repeats error texts

Reply:
fixed